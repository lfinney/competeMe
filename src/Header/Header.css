.Header {
  height: 100px;
  position: sticky;
  top: 0px;
  color: #6da34d;
  background-color: #1a1a1a;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  align-content: center;
  grid-area: top;
  z-index: 1; }

.Header .headerPrompt {
  display: flex;
  justify-content: space-around;
  align-items: center;
  width: 80%; }

.Header h1 {
  margin: 0;
  position: absolute;
  top: 15px;
  left: 20px;
  font-size: 3.5em; }

.Header h2 {
  margin: 0; }

.Header .welcome-message {
  display: inline-block;
  margin: auto; }

.Header button {
  margin-right: 50px;
  font-size: 1em;
  font-weight: bold;
  margin-left: auto;
  border: none;
  background-color: #6da34d;
  color: #393939;
  width: 100px;
  height: 50px;
  border-radius: 10px; }
  .Header button:hover {
    background-color: #393939; }
  .Header button:hover {
    color: #6da34d; }
