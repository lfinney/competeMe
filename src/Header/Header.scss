@import '../variables.scss';

.Header {
  background: $black;
  background: -moz-linear-gradient(top, $gray-black 0%, $black 100%);
  background: -webkit-linear-gradient(top, $gray-black 0%, $black 100%);
  background: linear-gradient(to bottom, $gray-black 0%, $black 100%);
  height: 100px;
  position: sticky;
  top: 0px;
  color: $grass;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  align-content: center;
  grid-area: top;
  z-index: 1;
}

.Header .headerPrompt {
  display: flex;
  justify-content: space-around;
  align-items: center;
  width: 80%;
}

.Header h1 {
  margin: 0;
  position: absolute;
  top: 15px;
  left: 20px;
  font-size: 3.5em;
}

.Header h2 {
  margin: 0;
}

.Header .welcome-message {
  display: inline-block;
  margin-left: auto;
  width: 60%;
}

.Header .header-text {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
}

.Header button {
  margin-right: 50px;
  font-size: 1em;
  font-weight: bold;
  margin-left: auto;
  border: $border;
  background: $black;
  background: -moz-linear-gradient(top, $secondary-two 0%, $primary-one 100%);
  background: -webkit-linear-gradient(top, $secondary-two 0%, $primary-one 100%);
  background: linear-gradient(to bottom, $secondary-two 0%, $primary-one 100%);
    &:hover {
      background: $black;
      background: -moz-linear-gradient(top, $gray-black 0%, $black 100%);
      background: -webkit-linear-gradient(top, $gray-black 0%, $black 100%);
      background: linear-gradient(to bottom, $gray-black 0%, $black 100%);
    }
  color: $black;
    &:hover {
      color: $primary-one;
    }
  width: 100px;
  height: 50px;
  border-radius: $border-radius;
}
