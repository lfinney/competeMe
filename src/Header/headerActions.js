export const activeUser = (user) => {
  return {
    type: 'USER_LOGIN',
    user
  };
};
