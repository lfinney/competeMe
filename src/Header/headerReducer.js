export function activeUser(state = {}, action) {
  switch (action.type) {
  case 'USER_LOGIN':
    return action.user;

  case 'UPDATE_USER_COMPS':
    return Object.assign({}, action.activeUser, {competitions: [...action.joinedCompetitions]});

  case 'USER_COMPETITON':
    return Object.assign({}, action.user, {competitions: action.comp});
  default:
    return state;
  }
}
