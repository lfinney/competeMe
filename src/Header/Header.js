import React, { Component } from 'react';
import { connect } from 'react-redux';
import { activeUser } from './headerActions';
import PropTypes from 'prop-types';
import firebase, { auth, provider } from '../firebase.js';
import Popup from '../Popup/Popup';

export class Header extends Component {
  componentDidMount() {
    const usersRef = firebase.database().ref('users');
    console.log(usersRef);

    auth.onAuthStateChanged((user) => {
      if (user) {
        const cleanedUser = this.cleanUserData(user);
        this.props.activeUser(cleanedUser);
      }
    });
  }

  cleanUserData(user) {
    return {
      userId: user.uid,
      displayName: user.displayName.split(' ')[0],
      email: user.email,
      competitions: []
    };
  }

  login = () => {
    auth.signInWithPopup(provider)
      .then((result) => result = result.user)
      .then((userData) => {
        const cleanedUser = this.cleanUserData(userData);
        this.props.activeUser(cleanedUser);
      });
    // firebase.auth().onAuthStateChanged( () => )
  }

  logout = () => {
    auth.signOut()
      .then(() => {
        this.props.activeUser({});
      });
  }

  render () {
    return (
      <div className="Header">
        <h1>CompeteMe</h1>
        {this.props.liveUser ?
          <div className="headerPrompt">
            <h2>Ready to get your game on, {this.props.user.displayName}?</h2>
            <p>Join or create a game below.</p>
            <button onClick={this.logout}>Log Out</button>
          </div>
          :
          <div className="headerPrompt">
            <h2>Sign-in using Google to join the competitions!</h2>
            <button onClick={this.login}>Log In</button>
          </div>
        }
        {
          this.props.activePopup && <Popup />        }
      </div>
    );
  }
}

Header.propTypes = {
  activeUser: PropTypes.func,
  user: PropTypes.object,
  liveUser: PropTypes.bool
};

const mapStateToProps = (store) => ({
  liveUser: store.activeUser.userId ? true : false,
  user: store.activeUser,
  activePopup: store.activePopup
});

const mapDispatchToProps = (dispatch) => ({
  activeUser: ( user ) => { dispatch(activeUser(user)); }
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
