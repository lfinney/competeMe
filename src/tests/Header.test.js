import React from 'react';
import ReactDOM from 'react-dom';
import Header from '../Header/Header';
import { shallow } from 'enzyme';


describe('Header', () => {

  it('should render component', () => {
    const wrapper = shallow(
      <Header />);

    expect(wrapper.length).toEqual(1);
  });
});
