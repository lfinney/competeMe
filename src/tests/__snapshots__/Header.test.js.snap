// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          CompeteMe
</h1>,
        <div
          className="headerPrompt"
>
          <h2
                    className="welcome-message"
          >
                    Sign-in using Google to join the competitions!
          </h2>
          <button
                    onClick={[Function]}
          >
                    Log In
          </button>
</div>,
        undefined,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "CompeteMe",
        },
        "ref": null,
        "rendered": "CompeteMe",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="welcome-message"
>
              Sign-in using Google to join the competitions!
</h2>,
            <button
              onClick={[Function]}
>
              Log In
</button>,
          ],
          "className": "headerPrompt",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Sign-in using Google to join the competitions!",
              "className": "welcome-message",
            },
            "ref": null,
            "rendered": "Sign-in using Google to join the competitions!",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Log In",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Log In",
            "type": "button",
          },
        ],
        "type": "div",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            CompeteMe
</h1>,
          <div
            className="headerPrompt"
>
            <h2
                        className="welcome-message"
            >
                        Sign-in using Google to join the competitions!
            </h2>
            <button
                        onClick={[Function]}
            >
                        Log In
            </button>
</div>,
          undefined,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "CompeteMe",
          },
          "ref": null,
          "rendered": "CompeteMe",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="welcome-message"
>
                Sign-in using Google to join the competitions!
</h2>,
              <button
                onClick={[Function]}
>
                Log In
</button>,
            ],
            "className": "headerPrompt",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Sign-in using Google to join the competitions!",
                "className": "welcome-message",
              },
              "ref": null,
              "rendered": "Sign-in using Google to join the competitions!",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Log In",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Log In",
              "type": "button",
            },
          ],
          "type": "div",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
